src/components/auth/SetupForm.tsx

```diff
@@ -0,0 +1,125 @@
+import { useTranslation } from 'react-i18next';
+import { Button, Checkbox, Dropdown, Form, Input } from 'semantic-ui-react';
+import React, { ChangeEvent, useEffect, useRef, useState } from 'react';
+import { Gender } from '../../clients/server.generated';
+import { Dispatch } from 'redux';
+import { connect } from 'react-redux';
+import { authSetup } from '../../stores/auth/actionCreators';
+import { useNavigate } from 'react-router-dom';
+import { NavigateFunction } from 'react-router/dist/lib/hooks';
+
+interface Props {
+  setup: (
+    email: string,
+    firstname: string,
+    lastname: string,
+    gender: string,
+    password: string,
+    rememberMe: boolean,
+    navigate: NavigateFunction) => void;
+}
+
+function SetupForm(props: Props) {
+  const { t } = useTranslation();
+  const [email, changeEmail] = useState('');
+  const [firstName, changeFirstName] = useState('');
+  const [lastName, changeLastName] = useState('');
+  const [gender, changeGender] = useState(Gender.UNKNOWN);
+  const [password, changePassword] = useState('');
+  const [rememberMe, changeRememberMe] = useState(false);
+
+  const inputRef = useRef<Input>(null);
+  useEffect(() => {
+    inputRef.current!.focus();
+  }, []);
+
+  const navigate = useNavigate();
+
+  return (
+    <Form siz={'large'}>
+      <Form.Field>
+        <Input
+          value={email}
+          placeholder={t('pages.setup.email')}
+          onChange={(e: ChangeEvent<HTMLInputElement>) => changeEmail(e.target.value)}
+          ref={inputRef}
+        />
+      </Form.Field>
```

Comment:
Please add input validation to this field, because I can fill in anything. It should be a valid email address.